src/audio/asrc/asrc.c

```diff
@@ -1080,19 +1035,31 @@ static int asrc_copy(struct comp_dev *dev)
 		cd->source_frames = MIN(cd->source_frames, frames_src);
 	}
 
-	if (cd->source_frames && cd->sink_frames)
-		asrc_process(dev, source, sink);
+	if (cd->source_frames && cd->sink_frames) {
+		int consumed = 0;
+		int produced = 0;
+
+		/* consumed bytes are not known at this point */
+		buffer_stream_invalidate(source, audio_stream_get_size(source_s));
+		cd->asrc_func(mod, source_s, sink_s, &consumed,
+				&produced);
```

Comment:
since you'll be splitting this commit into two: first adding `.trigger()` to module-adapter and then the actual ASRC conversion, you could also merge this into one line while at it